test/audience.test.ts

```diff
@@ -0,0 +1,122 @@
+import { edkt } from '../src';
+import { AudienceDefinition } from '../types';
+
+const sportKeywords = ['golf', 'liverpool', 'football', 'sport'];
+
+const sportPageFeatureGetter = {
+  name: 'keywords',
+  func: (): Promise<string[]> => {
+    return Promise.resolve(sportKeywords);
+  },
+};
+
+const sportAudienceDefinition: AudienceDefinition = {
+  id: 'sport_id',
+  name: 'Sport Audience',
+  ttl: 1,
+  lookback: 2,
+  occurrences: 3,
+  keywords: ['sport'],
+};
+
+describe('Test edkt audience matching', () => {
+  beforeAll(async () => {
+    localStorage.clear();
+    // add one initial view
+    await edkt.run({
+      pageFeatureGetters: [sportPageFeatureGetter],
+      audienceDefinitions: [sportAudienceDefinition],
+    });
+  });
+
+  it('First run -> add page view but do not match', async () => {
+    await edkt.run({
+      pageFeatureGetters: [sportPageFeatureGetter],
+      audienceDefinitions: [sportAudienceDefinition],
+    });
+
+    const edktPageViews = JSON.parse(
+      localStorage.getItem('edkt_page_views') || '[]'
+    );
+
+    const edktMatchedAudeinces = JSON.parse(
+      localStorage.getItem('edkt_matched_audiences') || '[]'
+    );
+
+    expect(edktPageViews.length).toEqual(2);
+    expect(edktMatchedAudeinces.length).toEqual(0);
+  });
+
+  it('Second run -> add another page view but do not match', async () => {
+    await edkt.run({
+      pageFeatureGetters: [sportPageFeatureGetter],
+      audienceDefinitions: [sportAudienceDefinition],
+    });
+
+    const edktPageViews = JSON.parse(
+      localStorage.getItem('edkt_page_views') || '[]'
+    );
+
+    const edktMatchedAudiences = JSON.parse(
+      localStorage.getItem('edkt_matched_audiences') || '[]'
+    );
+
+    expect(edktPageViews.length).toEqual(3);
+    expect(edktMatchedAudiences.length).toEqual(0);
+  });
+
+  it('Third run -> add another page view & now match', async () => {
```

Comment:
What do you guys think of having a loop for this instead since the previous tests are all similar? something like:
```js
for (let i = 0; i <= sportAudienceDefinition.occurrences; i++) {
  it(`Should ${i == sportAudienceDefinition.occurrences ? 'match' : 'not match'} at ${i + 1} page views`, async () => {
   ...
  })
}
```

It's a minor bikeshed but it might help improve understanding of what the tests are doing, since it's somewhat easy to miss that this matching is tied to `sportAudienceDefinition.occurrences`

Reply:
@Risto-Stevcev I would tend to avoid having loops to create the tests. because in this case it would only provide two valuable tests. Match or not match. However the main test in the pr would be the run.test.ts The audience.test.ts was my first method of testing it, that I thought might be useful to leave in as boilerplate for the future and it does provide another test to the functionality